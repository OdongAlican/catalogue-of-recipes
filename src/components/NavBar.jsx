import React from 'react'

const NavBar = () => (
    <div data-testid="navbar" className="bg-secondary d-flex justify-content-center align-items-center p-4 text-white nav-bar-section">
        <h4>
            Stay Online
        </h4>
    </div>
)

export default NavBar