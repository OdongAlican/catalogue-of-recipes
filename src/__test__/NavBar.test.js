import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavBar />, div);
});

it('renders nav-bar correctly', () => {
  const { getAllByTestId } = render(<NavBar />);
  const result = getAllByTestId('navbar');
  expect(result[0]).toHaveTextContent('Stay Online');
});

it('another rendering of nav-bar correctly', () => {
  const { getAllByTestId } = render(<NavBar />);
  const result = getAllByTestId('navbar');
  expect(result[0]).not.toHaveTextContent('Stay Onlines');
});

it('matches navbar snapshot', () => {
  const tree = renderer.create(<NavBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
