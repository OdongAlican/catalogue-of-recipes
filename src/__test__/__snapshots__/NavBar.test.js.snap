// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches navbar snapshot 1`] = `
<div
  className="bg-secondary d-flex justify-content-center align-items-center p-4 text-white nav-bar-section"
  data-testid="navbar"
>
  <h4>
    Stay Online
  </h4>
</div>
`;
